desktop/core/src/desktop/lib/tasks/compress_files/compress_utils.py

```diff
@@ -39,7 +39,8 @@ def compress_files_in_hdfs(request, file_names, upload_path, archive_name):
 
   _upload_compress_files_script_to_hdfs(request.fs)
 
-  files = [{"value": upload_path + '/' + urllib_quote(file_name.encode('utf-8'), SAFE_CHARACTERS_URI)} for file_name in file_names]
+  files = [{"value": urllib_quote(upload_path.encode('utf-8'), SAFE_CHARACTERS_URI) + '/' + \
+    urllib_quote(file_name.encode('utf-8'), SAFE_CHARACTERS_URI)} for file_name in file_names]
```

Comment:
Curious: Same no change here? Although this might be related to task server so we can take a look that time.

Reply:
The difference here is that the `files` array is being used as input to the `shell_notebook.add_shell_snippet` and not as response to the browser request. The `compress_files_in_hdfs.sh` function currently requires the files values to be encoded.